### PR TITLE
Add settings with live changes

### PR DESCRIPTION
**Pull request by MyBlackMIDIScore**

### CODE REVIEW

File: src/parsers.rs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+        Ok(())
+    }
+
+    fn create_empty(path: &PathBuf) -> Result<(), String> {
+        Self::default().save(path)
+    }
+}
+
+pub trait ConfigPub: Config {
```
Comment (arduano) [inline src/parsers.rs:38]: I would reverse the names personally, Config and ConfigInternals

Comment (arduano) [reply src/parsers.rs:38]: You could also `impl<T: Config> ConfigPub for T`, then you don't need any `impl ConfigPub for Settings {}`

File: src/parsers.rs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+mod soundfonts;
+pub use soundfonts::*;
+
+mod settings;
+pub use settings::*;
+
+const CONFIG_DIR: &str = "xsynth-kdmapi";
+
+pub trait Config: Default + Serialize + for<'a> Deserialize<'a> {
```
Comment (arduano) [inline src/parsers.rs:12]: Tbh reading the complexity of this more, I feel like most of this should just be a struct instead, like a `Config<T>` where T is Settings or SfList

Comment (arduano) [reply src/parsers.rs:12]: then you can have all the same methods but without the trait complexity

Comment (arduano) [reply src/parsers.rs:12]: You can pass in the paths via a constructor argument, makes things simpler

**Review by arduano — CHANGES_REQUESTED** [review]

File: src/parsers.rs (hunk 2 above)
Comment (MyBlackMIDIScore) [reply src/parsers.rs:12]: That does sound better yeah

File: src/parsers.rs (hunk 1 above)
Comment (MyBlackMIDIScore) [reply src/parsers.rs:38]: > You could also `impl<T: Config> ConfigPub for T`, then you don't need any `impl ConfigPub for Settings {}`

Oh that's actually nice I didn't know about that

**Pull request merged**